### DISCUSSION TITLE
Using rg inside mounted dir via sshfs.

### DISCUSSION THREAD

**Discussion by ApplicationDeveloper** [BurntSushi/ripgrep#1619] [General] [answered]
I use ripgrep in fzf. It works awesome.
Is it expected that rg --files is slow when used in a mounted dir using sshfs?
Or is it just because of ssh?
Or is there an option that I have to use that I overlooked?

**Comment by BurntSushi** [comment, marked as answer]
In _absolute_ terms, absolutely, it is expected. Consider what `sshfs` is doing: instead of reading file metadata from your local disk, it has to read that metadata over a network. Depending on your network latency, this could be significantly slower.

If you run `rg --files` twice in a row on an sshfs mounted directory, the second run may be faster due to the file metadata being cached in memory on your local system. For this reason, it's hard to do proper speed comparisons. If you're on Linux, then you can clear your local system's file cache with the following command:

```
$ sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'
```

That should then cause `rg --files` to run more slowly again.

The more relevant comparison here is whether `rg --files` is similar in speed to, say, something like `find ./ -type f`. They aren't doing strictly the same thing (since ripgrep respects your gitignore rules), but they should be similarish. You could use `rg -uuu --files` for a more fair comparison. Just remember to clear your file cache, as suggested above, between runs.

> Or is there an option that I have to use that I overlooked?

There is no option. ripgrep doesn't care about sshfs. It just uses (and minimizes the number of) normal file stat calls to get metadata and read directory entries.
